src/posthog-surveys.ts

```diff
@@ -214,34 +216,63 @@ export class PostHogSurveys {
         }
 
-        if (this._surveyEventReceiver == null) {
-            this._surveyEventReceiver = new SurveyEventReceiver(this.instance)
-        }
+        this._isInitializingSurveys = true
 
-        if (!generateSurveys) {
-            const loadExternalDependency = phExtensions.loadExternalDependency
+        try {
+            const generateSurveys = phExtensions.generateSurveys
 
-            if (loadExternalDependency) {
-                loadExternalDependency(this.instance, 'surveys', (err) => {
-                    if (err) {
-                        logger.error('Could not load surveys script', err)
-                        return
-                    }
+            if (!this._decideServerResponse) {
+                logger.warn('Decide not loaded yet. Not loading surveys.')
+                this._isInitializingSurveys = false
+                return
+            }
 
-                    this._surveyManager = phExtensions.generateSurveys?.(this.instance)
-                })
+            if (this._surveyEventReceiver == null) {
+                this._surveyEventReceiver = new SurveyEventReceiver(this.instance)
+            }
+
+            if (!generateSurveys) {
+                const loadExternalDependency = phExtensions.loadExternalDependency
+
+                if (loadExternalDependency) {
+                    loadExternalDependency(this.instance, 'surveys', (err) => {
+                        if (err) {
+                            logger.error('Could not load surveys script', err)
+                            this._isInitializingSurveys = false
+                            return
+                        }
+
+                        try {
+                            if (!this._surveyManager) {
+                                // Double-check we still don't have a manager
+                                this._surveyManager = phExtensions.generateSurveys?.(this.instance)
+                            }
+                        } finally {
+                            this._isInitializingSurveys = false
+                        }
+                    })
+                } else {
+                    logger.error('PostHog loadExternalDependency extension not found. Cannot load remote config.')
+                    this._isInitializingSurveys = false
+                }
             } else {
-                logger.error('PostHog loadExternalDependency extension not found. Cannot load remote config.')
+                try {
+                    if (!this._surveyManager) {
+                        // Double-check we still don't have a manager
+                        this._surveyManager = generateSurveys(this.instance)
+                    }
+                } finally {
+                    this._isInitializingSurveys = false
+                }
             }
-        } else {
-            this._surveyManager = generateSurveys(this.instance)
+        } catch (e) {
+            this._isInitializingSurveys = false
```

Comment:
should this be within the finally block instead?

Reply:
I agree, and you can then remove all of the inner `this._isInitializingSurveys = false` calls

Reply:
agreed, made the change